### PR TITLE
 [xdoctest][task 125-137] reformat example code with google style in static/*

### PR DESCRIPTION
**Pull request by DrRyanHuang**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

任务125-137

序号 | 行数 | 文件
-- | -- | --
|125 | 129 | python/paddle/static/amp/bf16/amp_lists.py |   |  
|126 | 623 | python/paddle/static/amp/bf16/amp_utils.py |   |  
|127 | 335 | python/paddle/static/amp/bf16/decorator.py |   |  
|128 | 286 | python/paddle/static/amp/debugging.py |   |  
|129 | 945 | python/paddle/static/amp/decorator.py |   |  
|130 | 899 | python/paddle/static/amp/fp16_utils.py |   |  
|131 | 371 | python/paddle/static/input.py |   |  
|132 | 1925 | python/paddle/static/io.py |   |  
|133 | 53 | python/paddle/static/log_helper.py |   |  
|134 | 4001 | python/paddle/static/nn/common.py |   |  
|135 | 1435 | python/paddle/static/nn/control_flow.py |   |  
|136 | 264 | python/paddle/static/nn/loss.py |   |  
|137 | 569 | python/paddle/static/nn/metric.py |  |  

<br class="Apple-interchange-newline">

任务125-137中还缺少

- `python/paddle/static/nn/common.py`
- `python/paddle/static/nn/control_flow.py`

预览：

- en: http://preview-paddle-pr-56121.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/abs_cn.html
- zh: http://preview-paddle-pr-56121.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/abs_cn.html

### Related links
 
- #55629
- #55295

### CODE REVIEW

File: python/paddle/static/amp/bf16/amp_lists.py [hunk 1]
```diff
@@ -34,10 +34,10 @@ class AutoMixedPrecisionListsBF16:
...
     Examples:
         .. code-block:: python
-        import paddle
-        paddle.enable_static()
-        with paddle.static.amp.bf16_guard():
-            paddle.static.amp.bf16.AutoMixedPrecisionListsBF16(custom_fp32_list={'lstm'})
+        >>> import paddle
+        >>> paddle.enable_static()
+        >>> with paddle.static.amp.fp16_guard():
+        ...     paddle.static.amp.bf16.AutoMixedPrecisionListsBF16(custom_fp32_list={'lstm'})
```
Comment (megemini) [inline python/paddle/static/amp/bf16/amp_lists.py]: 代码与 `.. code-block:: python` 要有缩进

File: python/paddle/static/amp/bf16/decorator.py [hunk 2]
```diff
@@ -267,59 +267,59 @@ def decorate_bf16(
...
+            >>> data = static.data(name='X', shape=[None, 1], dtype='float32')
+            >>> hidden = static.nn.fc(x=data, size=10)
+            >>> loss = paddle.mean(hidden)
+            >>> optimizer = paddle.optimizer.Adam(learning_rate=0.001)
 
-            mp_optimizer = static.amp.decorate_bf16(optimizer=optimizer)
+            >>> mp_optimizer = static.amp.bf16.decorate_bf16(optimizer=optimizer)
 
-            ops, param_grads = mp_optimizer.minimize(loss)
+            >>> ops, param_grads = mp_optimizer.minimize(loss)
```
Comment (megemini) [inline python/paddle/static/amp/bf16/decorator.py]: 要有缩进

File: python/paddle/static/amp/debugging.py [hunk 3]
```diff
@@ -205,37 +205,37 @@ def collect_operator_stats(program=None, print_subblocks=False):
 
      .. code-block:: python
```
Comment (megemini) [inline python/paddle/static/amp/debugging.py]: `.. code-block:: python` 与前面的 `Examples:` 也要有4个缩进～

File: python/paddle/static/amp/debugging.py [hunk 4]
```diff
@@ -205,37 +205,37 @@ def collect_operator_stats(program=None, print_subblocks=False):
...
+        ...         out = model(x)
+        ...         loss = paddle.mean(out)
+        ...         optimizer = paddle.optimizer.AdamW()
+        ...         optimizer = paddle.static.amp.decorate(optimizer)
+        ...         optimizer.minimize(loss)
+        >>> paddle.static.amp.debugging.collect_operator_stats(main_program)
+
         # <------------------------------------------------ op list of all blocks ------------------------------------------------->
         # <------------------------------------------------------- op list -------------------------------------------------------->
         # <--------------- Op Name ---------------- | -- FP16 Calls --- | -- BF16 Calls --- | --- FP32 Calls--- | -- Other Calls -->
```
Comment (megemini) [inline python/paddle/static/amp/debugging.py]: 这部分是输出，可以把前面的 `#` 去掉 ~

File: python/paddle/static/amp/decorator.py [hunk 5]
```diff
@@ -716,66 +717,67 @@ def decorate(
             .. code-block:: python
 
             # black&white list based strategy example
```
Comment (megemini) [inline python/paddle/static/amp/decorator.py:720]: 要有缩进

File: python/paddle/static/input.py [hunk 6]
```diff
@@ -52,38 +52,46 @@ def data(name, shape, dtype=None, lod_level=0):
...
+            # np-ndarray of shape=[3, 2, 1], dtype=float32, whose elements are 2
+            >>> print(out)
+            [array([[[2.],
+                    [2.]],
+
+                [[2.],
+                    [2.]],
+
+                [[2.],
+                    [2.]]], dtype=float32)]
```
Comment (megemini) [inline python/paddle/static/input.py]: 输出不要留空行

File: python/paddle/static/amp/decorator.py [hunk 7]
```diff
@@ -861,45 +863,45 @@ def decorate(  # noqa: F811
 
      .. code-block:: python
```
Comment (megemini) [inline python/paddle/static/amp/decorator.py]: 注意与 `Examples` 的缩进

File: python/paddle/static/io.py [hunk 8]
```diff
@@ -1732,14 +1743,14 @@ def get_program_persistable_vars(program):
...
         .. code-block:: python
-            import paddle
-            import paddle.static.io as io
-            import paddle.fluid as fluid
-            paddle.enable_static()
-            data = paddle.static.data(name="img", shape=[64, 784])
-            w = paddle.create_parameter(shape=[784, 200], dtype='float32', name='fc_w')
-            b = paddle.create_parameter(shape=[200], dtype='float32', name='fc_b')
-            list_para  = io.get_program_persistable_vars(  fluid.default_main_program() )
+            >>> import paddle
```
Comment (megemini) [inline python/paddle/static/io.py:1810]: 留个空行

File: python/paddle/static/nn/metric.py [hunk 9]
```diff
@@ -175,53 +177,61 @@ def auc(
             stat_neg: the statistic for label=0 at the time of calculation
 
 
-    Examples:
+    Examples 1:
```
Comment (megemini) [inline python/paddle/static/nn/metric.py]: 不要这么写 ～

如果有多个例子，用 `:name: XXX` 来区分：

``` python
Examples:

    .. code-block:: python
        :name: example-XXX

        >>> xxx

```

**Review by megemini — COMMENTED** [review]
@SigureMo static 里面太多重复 name 的变量了，导致很多示例没法跑 ～

static 处理 name 在 scope.h/scope.cc 里面，python 这边还不好 hack 。

看看怎么处理？跳过？重命名？

还有，这个 name 在底层可以为 None：

``` python
    def create_var(self, *args, **kwargs):
        if in_dygraph_mode():
            var = _create_tensor(*args, **kwargs)
        else:
            var = Variable(block=self, *args, **kwargs)
            if 'initializer' in kwargs:
                kwargs['initializer'](var, self)
        return var

```

这里 `_create_tensor` 和 `Variable` 的 name 都可以是 None，可是 `data` 方法却限制不能为 None ～

**Comment by SigureMo** [comment]
> @SigureMo static 里面太多重复 name 的变量了，导致很多示例没法跑 ～

这个回复之前没注意到，这个如果改成多进程是不是能解决呢？ @megemini

**Comment by megemini** [comment]
> 这个回复之前没注意到，这个如果改成多进程是不是能解决呢？ @megemini

我测试的可以 ～ 所以先更新了 convert-doctest 工具，可以验证一下可行性。

**Comment by megemini** [comment]
@DrRyanHuang 不好意思，这个 PR 可能要等一下，xdoctest 没法做到环境隔离，导致 static 的一些示例不能跑 ～

等这个问题解决了，可以重新跑一下 CI ～ 不好意思 ～ 😅

File: python/paddle/static/amp/bf16/decorator.py [hunk 10]
```diff
@@ -264,7 +264,7 @@ def decorate_bf16(
         enabled.
 
     Examples 1:
```
Comment (SigureMo) [inline python/paddle/static/amp/bf16/decorator.py]: 这里的 `Examples 1`、`Examples 2` 也要改一下

**Comment by SigureMo** [comment]
#56400 已 merge，麻烦 merge 下最新的 develop 试试～

File: python/paddle/static/amp/debugging.py [hunk 11]
```diff
@@ -203,65 +203,65 @@ def collect_operator_stats(program=None, print_subblocks=False):
...
+          reduce_mean                             |  0                |  0                |  1                |  0
+          reduce_mean_grad                        |  0                |  0                |  1                |  0
+          relu                                    |  1                |  0                |  0                |  0
+          relu_grad                               |  1                |  0                |  0                |  0
+          reshape2                                |  0                |  0                |  1                |  0
+          reshape2_grad                           |  0                |  0                |  1                |  0
+          softmax                                 |  0                |  0                |  1                |  0
+          softmax_grad                            |  0                |  0                |  1                |  0
+          update_loss_scaling                     |  0                |  0                |  1                |  0
+        <----------------------------------------------------- op count: 22 ----------------------------------------------------->
```
Comment (megemini) [inline python/paddle/static/amp/debugging.py]: 238这行开始输出，不要留这个空行～ 另外，这里缩进对齐一下上面的 `>>>`

File: python/paddle/static/amp/decorator.py [hunk 12]
```diff
@@ -713,69 +714,70 @@ def decorate(
...
+            >>> mp_optimizer = static.amp.decorate(
+            ...         optimizer=optimizer, init_loss_scaling=8.0)
 
-            ops, param_grads = mp_optimizer.minimize(loss)
-            scaled_loss = mp_optimizer.get_scaled_loss()
+            >>> ops, param_grads = mp_optimizer.minimize(loss)
+            >>> scaled_loss = mp_optimizer.get_scaled_loss()
 
     Examples 2:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/static/amp/decorator.py]: 这里跟之前一样，改为 `Examples:` `:name:` 的形式 ～

File: python/paddle/static/nn/metric.py [hunk 13]
```diff
@@ -353,23 +365,22 @@ def ctr_metric_bundle(input, label, ins_tag_weight=None):
...
-            predict = paddle.nn.functional.sigmoid(paddle.static.nn.fc(input=data, size=1))
-            auc_out = paddle.static.ctr_metric_bundle(input=predict, label=label)
+            >>> import paddle
+            >>> paddle.enable_static()
+            >>> data = paddle.static.data(name="data", shape=[-1, 32], dtype="float32")
+            >>> label = paddle.static.data(name="label", shape=[-1, 1], dtype="int32")
+            >>> predict = paddle.nn.functional.sigmoid(paddle.static.nn.fc(x=data, size=1))
+            >>> auc_out = paddle.static.ctr_metric_bundle(input=predict, label=label)
     Examples 2:
         .. code-block:: python
```
Comment (megemini) [inline python/paddle/static/nn/metric.py]: 改为 `Examples:` 和 `:name:` 的形式 ～

File: python/paddle/static/io.py [hunk 14]
```diff
@@ -1001,36 +1006,40 @@ def save_vars(
...
+            >>> main_prog = fluid.Program()
+            >>> startup_prog = fluid.Program()
+            >>> with fluid.program_guard(main_prog, startup_prog):
+            ...     data = paddle.static.data(name="img", shape=[64, 784])
+            ...     w = paddle.create_parameter(shape=[784, 200], dtype='float32', name='fc_w')
+            ...     b = paddle.create_parameter(shape=[200], dtype='float32', name='fc_b')
+            ...     hidden_w = paddle.matmul(x=data, y=w)
+            ...     hidden_b = paddle.add(hidden_w, b)
+            >>> place = fluid.CPUPlace()
+            >>> exe = fluid.Executor(place)
```
Comment (megemini) [inline python/paddle/static/io.py]: 不要用 fluid 了～

File: python/paddle/static/io.py [hunk 15]
```diff
@@ -1154,41 +1163,43 @@ def load_vars(
...
```
Comment (megemini) [inline python/paddle/static/io.py]: 修改 fluid

File: python/paddle/static/io.py [hunk 16]
```diff
@@ -1795,14 +1806,15 @@ def get_program_persistable_vars(program):
...
-            list_para  = io.get_program_persistable_vars(  fluid.default_main_program() )
+
+            >>> import paddle
+            >>> import paddle.static.io as io
+            >>> import paddle.fluid as fluid
+            >>> paddle.enable_static()
+            >>> data = paddle.static.data(name="img", shape=[64, 784])
+            >>> w = paddle.create_parameter(shape=[784, 200], dtype='float32', name='fc_w')
+            >>> b = paddle.create_parameter(shape=[200], dtype='float32', name='fc_b')
+            >>> list_para  = io.get_program_persistable_vars(  fluid.default_main_program() )
```
Comment (megemini) [inline python/paddle/static/io.py]: 修改 fluid ～

File: python/paddle/static/amp/bf16/amp_lists.py [hunk 17]
```diff
@@ -34,10 +34,11 @@ class AutoMixedPrecisionListsBF16:
...
     Examples:
         .. code-block:: python
-        import paddle
-        paddle.enable_static()
-        with paddle.static.amp.bf16_guard():
-            paddle.static.amp.bf16.AutoMixedPrecisionListsBF16(custom_fp32_list={'lstm'})
+
+            >>> import paddle
+            >>> paddle.enable_static()
+            >>> with paddle.static.amp.fp16_guard():
```
Comment (megemini) [inline python/paddle/static/amp/bf16/amp_lists.py]: `with paddle.static.amp.fp16_guard():` -> `with paddle.static.amp.bf16.bf16_guard():`

这里是 bf16 的示例代码 ～

File: python/paddle/static/amp/bf16/amp_utils.py [hunk 18]
```diff
@@ -230,18 +230,18 @@ def bf16_guard():
...
-                hidden = paddle.static.nn.fc(pool, size=10)
-                loss = paddle.mean(hidden)
+            >>> import numpy as np
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+            >>> paddle.enable_static()
+            >>> data = paddle.static.data(name='X', shape=[None, 1, 28, 28], dtype='float32')
+            >>> conv2d = paddle.static.nn.conv2d(input=data, num_filters=6, filter_size=3)
+
+            >>> with paddle.static.amp.fp16_guard():
```
Comment (megemini) [inline python/paddle/static/amp/bf16/amp_utils.py]: 同上

File: python/paddle/static/amp/bf16/decorator.py [hunk 19]
```diff
@@ -135,42 +135,42 @@ def amp_init(
...
+                >>> import paddle.nn.functional as F
+                >>> paddle.enable_static()
+
+                >>> def run_example_code():
+                ...     place = paddle.CPUPlace(0)
+                ...     exe = paddle.static.Executor(place)
+                ...     data = paddle.static.data(name='X', shape=[None, 1, 28, 28], dtype='float32')
+                ...     conv2d = paddle.static.nn.conv2d(input=data, num_filters=6, filter_size=3)
+                ...     # 1) Use bf16_guard to control the range of bf16 kernels used.
+                ...     with paddle.static.amp.bf16_guard():
```
Comment (megemini) [inline python/paddle/static/amp/bf16/decorator.py]: 同上，这里应该是 `with paddle.static.amp.bf16.bf16_guard():`

File: python/paddle/static/amp/bf16/decorator.py [hunk 20]
```diff
@@ -263,63 +263,67 @@ def decorate_bf16(
...
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> def run_example_code():
+            ...     place = paddle.CPUPlace(0)
+            ...     exe = paddle.static.Executor(place)
+            ...     data = paddle.static.data(name='X', shape=[None, 1, 28, 28], dtype='float32')
+            ...     conv2d = paddle.static.nn.conv2d(input=data, num_filters=6, filter_size=3)
+            ...     # 1) Use bf16_guard to control the range of bf16 kernels used.
+            ...     with paddle.static.amp.bf16_guard():
```
Comment (megemini) [inline python/paddle/static/amp/bf16/decorator.py]: 同上

File: python/paddle/static/amp/bf16/decorator.py [hunk 21]
```diff
@@ -263,63 +263,67 @@ def decorate_bf16(
...
+            ...     # 2) Create the optimizer and set `multi_precision` to True.
+            ...     # Setting `multi_precision` to True can avoid the poor accuracy
+            ...     # or the slow convergence in a way.
+            ...     optimizer = paddle.optimizer.Momentum(learning_rate=0.01, multi_precision=True)
+            ...     # 3) These ops in `custom_fp32_list` will keep in the float32 computation type.
+            ...     amp_list = paddle.static.amp.CustomOpLists(
+            ...         custom_fp32_list=['pool2d'])
+            ...     # 4) The entry of Paddle AMP.
+            ...     # Enable pure bf16 training by setting `use_pure_bf16` to True.
+            ...     optimizer = paddle.static.amp.decorate_bf16(
```
Comment (megemini) [inline python/paddle/static/amp/bf16/decorator.py]: 同上，paddle.static.amp.bf16.XXX

File: python/paddle/static/amp/decorator.py [hunk 22]
```diff
@@ -712,70 +713,73 @@ def decorate(
...
-                # If you want to perform the testing process, you should pass `test_program` into `amp_init`.
-                optimizer.amp_init(place, scope=paddle.static.global_scope())
-
-            if paddle.is_compiled_with_cuda() and len(paddle.static.cuda_places()) > 0:
-                run_example_code()
+            >>> import numpy as np
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> # doctest: +REQUIRES(env:GPU)
```
Comment (megemini) [inline python/paddle/static/amp/decorator.py]: 这里要加一个 `>>> paddle.enable_static()` ，不然会报错～

**Review by megemini — COMMENTED** [review]
东西很多，辛苦了！～～～

主要问题是，paddle.static.amp.bf16 里面的一些接口怎么给改为了 fp16？！🤣

另外，一些 examples 拆分的地方，相应的 docs 注意修改一下 ～

**Review by megemini — APPROVED** [review]
LGTM ～ 辛苦！

反馈几个文档问题

- ctr_metric_bundle 缺少中文文档
- paddle.static.auc 中文文档的示例没有分开

中文
image

英文
image

@SigureMo @sunzhongkai588

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾

> paddle.static.auc 中文文档的示例没有分开

麻烦 @DrRyanHuang 额外提一个中文文档 PR 利用 COPY-FROM 修复一下～

本 PR 可以合入

**Comment by sunzhongkai588** [comment]
> * ctr_metric_bundle 缺少中文文档

好的，这个我和api负责人反馈一下～
